### DISCUSSION TITLE
How to add q-select filter on q-table ?

### DISCUSSION THREAD

**Discussion by iamshulz** [quasarframework/quasar#9573] [General - Components / Directives / etc] [answered]
I have a q-table that has two template and one is for search filter which was already provided as a sample from the quasar docs.
Inside the template I have two filter one is for search specifically for keyword and one is for searching what type such as 'name','address' etc THAT is to be sent on axios.

```
<q-table
       title="Treats"
       style="height: 470px"
       class="patient-table-sticky-header"
       :rows="rows"
       :columns="columns"
       :loading="loading"
       :filter="filter"
       :filterType="filterType"
       :separator="separator"
       v-model:selected="selected"
       v-model:pagination="pagination"
       row-key="id"
       binary-state-sort
       selection="single"
       @request="onRequest"
>
                <!-- Search Filter & Search Type -->
                <template v-slot:top-left>
                    <q-input borderless dense debounce="1000" v-model="filter" placeholder="Search by Filter">
                        <template v-slot:append>
                            <q-icon name="search" />
                        </template>
                    </q-input>
                    <q-select borderless dense debounce="1000" v-model="filterType" :options="options" label="Filter" />
                </template>
 ( ..... )
</q-table>

<script>
onMounted(() => {
      // get initial data from server (1st page)
      onRequest({
         pagination: pagination.value,
         filter: undefined,
         filterType: undefined  // <--- I've already defined it here
      })
})
    
```

I have defined some values on select and it works, I'm able to show options
![image](https://user-images.githubusercontent.com/20100579/120898796-216e4680-c65f-11eb-900b-9a82f4524185.png)

However whenever I type some keword on search filter the prop "filterType" gets removed why is this?
![image](https://user-images.githubusercontent.com/20100579/120898761-e9670380-c65e-11eb-8c83-28b894ba0c9d.png)

Also another question not related with this. How should I know what to put on `v-slot: nameHere` how should I know `v-slot:top-left` exists or `v-slot:append` ?

**UPDATE**

I have found that I have to use `filter-method` however I have difficulty making it work with `q-select`

**Comment by ontwikkelfabriek** [comment]
> How should I know what to put on v-slot: nameHere how should I know v-slot:top-left exists or v-slot:append ?

Look at the  api docs of Q-table, it's under the `ScopedSlots (18)` section.
https://quasar.dev/vue-components/table#qtable-api

**Reply by iamshulz** [reply]
Thank you for pointing this out.

**Comment by iamshulz** [comment, marked as answer]
I got it working and it was really easy :( took me time to solve it. 

```
( ... )
const type = filterType.value.code === undefined ? '' : filterType.value.code
$store.dispatch('patient/requestPatientTable', { page, fetchCount, filter, sortBy, descending, type })
```
